includes/class-wc-subscription.php

```diff
@@ -2098,6 +2099,16 @@ public function get_last_order( $return_fields = 'ids', $order_types = array( 'p
 			}
 		}
 
+		if ( ! empty( $exclude_statuses ) ) {
+			$related_orders = array_filter(
+				$related_orders,
+				function( $order_id ) use ( $exclude_statuses ) {
+					$order = wc_get_order( $order_id );
+					return ! in_array( $order->get_status(), $exclude_statuses, true );
```

Comment:
We should use has status here as it's filterable and so we'd benefit from any status filtering that might occur. I've also changed it to make sure we don't cause a fatal if the order fails to load. 

```suggestion
					return $order && ! $order->has_status( $exclude_statuses );
```